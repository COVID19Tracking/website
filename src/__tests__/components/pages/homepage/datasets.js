import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import Datasets from '~components/pages/homepage/datasets'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    whiteHouse: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1.3,
          sizes: '(max-width: 1200px) 100vw, 1200px',
          src:
            '/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png',
          srcSet: '/srcset',
          tracedSVG: '[SVG]',
        },
      },
    },
    covidExit: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1.3,
          sizes: '(max-width: 1200px) 100vw, 1200px',
          src:
            '/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png',
          srcSet: '/srcset',
          tracedSVG: '[SVG]',
        },
      },
    },
    jhuTracker: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1.3,
          sizes: '(max-width: 1200px) 100vw, 1200px',
          src:
            '/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png',
          srcSet: '/srcset',
          tracedSVG: '[SVG]',
        },
      },
    },
  }))
})

describe('Components : Pages : Homepage : Datasets', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Datasets />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
