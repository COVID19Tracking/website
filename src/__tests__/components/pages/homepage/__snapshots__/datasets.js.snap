// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Homepage : Datasets renders correctly 1`] = `
<div>
  <div
    className="feature flip"
  >
    <div
      className="element"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "76.92307692307692%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="[SVG]"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1200px) 100vw, 1200px"
            srcSet="/srcset"
          />
          <img
            alt="Screenshot of a COVID Exit Strategy chart that uses COVID Tracking Project data in its state-by-state comparison of interventions, testing, and outcomes."
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1200px) 100vw, 1200px"
            src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
            srcSet="/srcset"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"Screenshot of a COVID Exit Strategy chart that uses COVID Tracking Project data in its state-by-state comparison of interventions, testing, and outcomes.\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="info"
    >
      <h3>
        Johns Hopkins
      </h3>
      <p>
        Johns Hopkins relies on our testing data for its
         
        <a
          href="https://coronavirus.jhu.edu/testing"
          rel="nofollow noopener noreferrer"
          target="_blank"
        >
          COVID-19 Testing Insights Initiative
        </a>
        , which brings data and expert analysis together in one place. The initiative is designed to help policymakers and the public understand the trajectory of the pandemic, and make decisions about the path forward.
      </p>
    </div>
  </div>
  <div
    className="feature"
  >
    <div
      className="element"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "76.92307692307692%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="[SVG]"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1200px) 100vw, 1200px"
            srcSet="/srcset"
          />
          <img
            alt="Image composite showing COVID Tracking Project data in use on a slide from the White House’s 'Opening Up America Again' testing strategy presentation deck."
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1200px) 100vw, 1200px"
            src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
            srcSet="/srcset"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"Image composite showing COVID Tracking Project data in use on a slide from the White House’s 'Opening Up America Again' testing strategy presentation deck.\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="info"
    >
      <h3>
        The White House
      </h3>
      <p>
        The White House chose the COVID Tracking Project as the best source to cite for daily US test numbers in its
         
        <a
          href="https://www.whitehouse.gov/wp-content/uploads/2020/04/Testing-Overview-Final.pdf"
        >
          “Opening Up America Again” testing strategy
        </a>
        .
      </p>
    </div>
  </div>
  <div
    className="feature flip"
  >
    <div
      className="element"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "76.92307692307692%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="[SVG]"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1200px) 100vw, 1200px"
            srcSet="/srcset"
          />
          <img
            alt="Screenshot of a COVID Exit Strategy chart that uses COVID Tracking Project data in its state-by-state comparison of interventions, testing, and outcomes."
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1200px) 100vw, 1200px"
            src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
            srcSet="/srcset"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"Screenshot of a COVID Exit Strategy chart that uses COVID Tracking Project data in its state-by-state comparison of interventions, testing, and outcomes.\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="info"
    >
      <h3>
        COVID Exit Strategy
      </h3>
      <p>
        Created by a group of public health and crisis experts,
         
        <a
          href="https://covidexitstrategy.org"
        >
          covidexitstrategy.org
        </a>
         
        identifies critical interventions needed to stop the spread of COVID-19, and urges government decision-makers to apply them. They use our data to power a dashboard comparing each state’s interventions and testing levels with case counts and deaths over time.
      </p>
    </div>
  </div>
</div>
`;
